src/master/manager.cpp

```diff
@@ -940,23 +860,82 @@ class ManagerProcess : public ProtobufProcess<ManagerProcess>
         agents.at(from.address.ip).updateOverlayState(message.overlays(i));
       }
 
-      // Update the Agent State.
-      const string agent = stringify(from.address.ip);
+      // We don't need to store the "state" of an overlay network on
+      // an Agent in the replicated log so go ahead and update the
+      // `networkState` without updating the overlay replicated log.
       for (int i = 0; i < networkState.agents_size(); i++) {
-        if (agent == networkState.agents(i).ip()) {
+        if (stringify(from.address.ip) == networkState.agents(i).ip()) {
           networkState.mutable_agents(i)->CopyFrom(
               agents.at(from.address.ip).getAgentInfo());
-          break;
+
+          send(from, AgentRegisteredAcknowledgement());
+          return;
         }
       }
-
-      send(from, AgentRegisteredAcknowledgement());
+      LOG(ERROR) << "Unable to find the registered agent in the `networkState`";
     } else {
       LOG(ERROR) << "Got ACK for network message for non-existent PID "
                  << from;
     }
   }
 
+  Future<http::Response> state(const http::Request& request)
+  {
+    VLOG(1) << "Responding to `state` endpoint";
+
+    return http::OK(
+        JSON::protobuf(networkState),
+        request.url.query.get("jsonp"));
+  }
+
+  void recover()
+  {
+    // Nothing to recover.
+    CHECK_NOTNULL(replicatedLog.get());
+
+    recovering = true;
+
+    replicatedLog->fetch<State>(REPLICATED_LOG_STORE_KEY)
+      .onAny(defer(self(),
+                   &ManagerProcess::_recover,
+                   lambda::_1));
+  }
+private:
+  bool recovering;
+  bool storing;
+
+  hashmap<string, Overlay> overlays;
+  hashmap<net::IP, Agent> agents;
+
+  Owned<mesos::state::protobuf::State> replicatedLog;
+
+  Option<Variable<State>> storedState;
```

Comment:
Add some comments to these variables, else people won't realize the difference between them.